examples/config-gc.json

```diff
@@ -3,10 +3,48 @@
     "storage": {
         "rootDirectory": "/tmp/zot",
         "gc": true,
-        "gcReferrers": true,
         "gcDelay": "2h",
         "untaggedImageRetentionDelay": "4h",
-        "gcInterval": "1h"
+        "gcInterval": "1h",
+        "retention": {
+            "dryRun": false,
+            "policies": [
+                {
+                    "repoNames": ["infra/*", "prod/*"],
+                    "deleteReferrers": false,
+                    "deleteUntagged": true,
+                    "tagsRetention": [{
+                        "names": ["v2.*", "*-prod"],
+                        "retainAlways": true
```

Comment:
Isnt this default behavior why do i need to configure a knob if am going to retain?

Reply:
because all the others tags will be removed, if you have a list of tagsRetention, only those matching its policies/rules will be retained, the rest will be removed.

Reply:
Now I understand the question, yes I think we can remove it.

Reply:
It is still unclear. The `names` key is optional right, if not present it matches all tags in the repo?
1. What happens if a tag retention entry has no keys at all, no "names", no "retainAlways", no other keys?
2. How do I specify a retention rule to keep everything in a repo? I used to think I need to add an entry having "retainAlways" as a single key with value "true". If this is not the case, do I need to use a regex to match any tag?

To clarify what I personally understand by "retain" being the default action:

1. If a repo matches the repo name pattern, but no retention rule, everything in that repo will be deleted correct?
Thinking the act of someone specifying a matcher for that repo is a commitment from that user that he wants the cleanup to be applied to that repo. So delete everything which does not match a tag retention rule for that repo.

2. If a repo doesn't match any repo name pattern, it means the user doesn't want to use this feature to clean up those repos, so nothing will be removed. -> this is my understanding of "retain by default".

Reply:
I removed that key.

1.1) it retains everything
1.2) if you want to keep everything for a repo you need to make sure your repo doesn't match any repo policy, in other words you don't write a policy for it.
or you put a default with repoNames: ["**"] where deleteReferrers is false, deleteUntagged is false, and no retention set.

2.1) if there are no retention rules at all, then it will retain everything, if there is at least one retention rule, and it doesn't match it then it will remove those tags that don't match

2.2) true, and it's true for the current logic.